docs/updating-the-docs.md

```diff
@@ -48,3 +46,26 @@ Examples are included in the markdown files using the [pymdown snippet notation]
 Make sure you commit both the changes to the `examples` and `databuilder/snippets` directories.
 
 If you are updating multiple examples it might be easier to the extraction step happen each time you change the python modules, you can do this with `just databuilder/watch-examples`.
+
+
+## Updating Data Builder backend and contract documentation
+
+!!! note "You will need both Python 3.7 and 3.10 on your PATH to make these updates"
+
+Some of the Data Builder documentation is generated from parts of databuilder's codebase.
```

Comment:
thought: I suspect there's more effort on the reader's part when trying to work out whether "Data Builder" and "databuilder" are the same or different here. My opinion rather than fact, though.

I think I'd always favour "Data Builder" unless explicitly referring to actually code use — that is, `import databuilder`. For the case of the repository, I'd probably write "Data Builder repository" and link to the repository. Anyone reading, feel free to suggest counter-examples.

(We may have inconsistencies elsewhere and we can tidy these up, before we make the documentation more public.)